### PR TITLE
tests: log hygiene checks for storage controller

### PR DESCRIPTION
**Pull request by jcsp**
## Problem

As with the pageserver, we should fail tests that emit unexpected log errors/warnings.

## Summary of changes

- Refactor existing log checks to be reusable
- Run log checks for attachment_service
- Add allow lists as needed.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Comment by jcsp** [comment]
not to self: one test_compatibility failure that I need to wait for CI to give me the allure report to figure out (`WARN process_result{tenant_id=95f60157a92a14e4164d9c75dab97c6d shard_id=0000 sequence=1}: Reconcile error: pageserver API: "no tenant_shard_id specified in path param"`)

File: test_runner/fixtures/neon_fixtures.py
Comment (VladLazar) [inline test_runner/fixtures/neon_fixtures.py:2362]: Did you remove the `scan_pageserver_log_for_errors` method? Maybe I'm not seeing it in the diff.

File: test_runner/regress/test_sharding_service.py
Comment (VladLazar) [inline test_runner/regress/test_sharding_service.py:850]: nit: s/permitr/permits

**Review by VladLazar — APPROVED** [review]

File: test_runner/fixtures/neon_fixtures.py
Comment (jcsp) [reply test_runner/fixtures/neon_fixtures.py:2362]: It's still used in the standalone execution mode of `allowed_errors.py`.  I'm asking in slack if that's actually used anywhere...

File: test_runner/regress/test_sharding_service.py
Comment (jcsp) [reply test_runner/regress/test_sharding_service.py:850]: updated (I was repushing this anyway to get a fresh run before merging)

File: test_runner/fixtures/neon_fixtures.py
Comment (VladLazar) [reply test_runner/fixtures/neon_fixtures.py:2366]: I think it might be used for benchmarks that run in staging. Not sure I see why it can't be removed and replaced with a call to the new method.

**Pull request merged**
